api/src/main/java/org/openmrs/api/ObsService.java

```diff
@@ -536,5 +537,32 @@ public Integer getObservationCount(List<Person> whom, List<Encounter> encounters
 			List<Concept> answers, List<PERSON_TYPE> personTypes, List<Location> locations, List<Visit> visits,
 			Integer obsGroupId, Date fromDate, Date toDate, boolean includeVoidedObs, String accessionNumber)
 			throws APIException;
-	
+
+	/**
+	 * This method gets ObsReferenceRange by id
+	 * 
+	 * @param id ObsReferenceRange id
+	 * @return {@link ObsReferenceRange}
+	 *
+	 * @since 2.7.0
+	 *
+	 * <strong>Should</strong> get the ObsReferenceRange with the given id
+	 * <strong>Should</strong> return null if ObsReferenceRange is not found
+	 */
+	@Authorized(PrivilegeConstants.GET_OBS)
+	ObsReferenceRange getObsReferenceRangeById(Integer id);
```

Comment:
Since the obs reference range is already a child of the obs, do we still need the above method?

Reply:
We needed this method if we have a different table for reference ranges. However, now that we are making these ranges part of Obs entity, we don't need it anymore.

Reply:
We still have a different table for obs reference ranges. But we load them as a property of the Obs object.